packages/dart/npt_flutter/lib/features/policy/cubit/policy_cubit.dart

```diff
@@ -0,0 +1,318 @@
+      // Don't switch if already viewing the same role
+      if (currentState.isRoleViewing && currentState.selectedRole?.id == roleId) {
+        return;
+      }
+      
+      final selectedRole = currentState.roles.firstWhere(
+        (role) => role.id == roleId,
+        orElse: () => currentState.roles.first,
+      );
+      
+      emit(currentState.copyWith(
+        selectedRole: selectedRole,
+        viewMode: PolicyViewMode.roleViewing,
+      ));
+    }
+  }
+
+  /// Start editing an existing role
+  void startEditingRole(String roleId) {
+    if (state is PolicyLoaded) {
+      final currentState = state as PolicyLoaded;
+      
+      final roleToEdit = currentState.roles.firstWhere(
+        (role) => role.id == roleId,
+        orElse: () => currentState.roles.first,
+      );
+      
+      emit(currentState.copyWith(
+        selectedRole: roleToEdit,
+        viewMode: PolicyViewMode.roleEditing,
+      ));
+    }
+  }
+
+  /// Start creating a new role
+  void startCreatingRole() {
+    if (state is PolicyLoaded) {
+      final currentState = state as PolicyLoaded;
+      
+      final emptyRole = Role.empty(name: '');
+      emit(currentState.copyWith(
+        selectedRole: emptyRole,
+        viewMode: PolicyViewMode.roleCreating,
+      ));
+    }
+  }
+
+  /// Cancel editing or creating and return to appropriate state
+  void cancelEditing() {
+    if (state is PolicyLoaded) {
+      final currentState = state as PolicyLoaded;
+      
+      if (currentState.isRoleCreating) {
+        // Return to browsing mode when canceling creation
+        emit(currentState.copyWith(
+          clearSelectedRole: true,
+          viewMode: PolicyViewMode.rolesBrowsing,
+        ));
+      } else if (currentState.isRoleEditing && currentState.hasSelectedRole) {
+        // Return to viewing mode when canceling edit
+        emit(currentState.copyWith(
+          viewMode: PolicyViewMode.roleViewing,
+        ));
+      }
+    }
+  }
+
+  /// Exit viewing mode and return to browsing
+  void exitViewing() {
+    if (state is PolicyLoaded) {
+      final currentState = state as PolicyLoaded;
+      
+      if (currentState.isRoleViewing) {
+        emit(currentState.copyWith(
+          clearSelectedRole: true,
+          viewMode: PolicyViewMode.rolesBrowsing,
+        ));
+      }
+    }
+  }
+
+  /// Create a new role
+  Future<void> createRole(Role role) async {
+    if (state is PolicyLoaded) {
+      final currentState = state as PolicyLoaded;
+      if (!currentState.isRoleCreating) return;
+      
+      emit(const PolicyLoading(operation: 'Creating role'));
+      
+      try {
+        final success = await _roleRepository.createNewRole(role);
+        if (success) {
+          // Refresh roles and show the created role
+          final updatedRoles = await _roleRepository.fetchRoles();
+          final createdRole = updatedRoles.firstWhere(
+            (r) => r.name == role.name && r.description == role.description,
```

Comment:
Use an id to ensure uniqueness of a role, rather than two user defined values, which could be duplicated across two entries.